docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,67 @@
+---
+title: NativeAOT deployment overview
+description: Learn what NativeAOT deployments are and why you should consider using it as part of the publishing your app with .NET 7 and later.
+author: lakshanf
+ms.author: lakshanf
+ms.date: 06/09/2022
+---
+# NativeAOT Deployment
+
+.NET NativeAOT deployment model uses an optimized ahead-of-time (AOT) compiler that generates an application that contains native executable code. NativeAOT applications do not use the just-in-time (JIT) compiler when the application runs. There are performance benefits (startup time, working set improvements etc.) and the ability to run .NET applications in restricted environment where a JIT is not allowed with using NativeAOT applications. NativeAOT applications have to be [self-contained applications](index.md#publish-self-contained) that targets a specific runtime environment (RID) such as Linux x64 or Windows x64.
+
+There are some limitations in .NET NativeAOT deployment model, the main one being, that no runtime code generation is possible. See [Impact of using the NativeAOT deployment](#impact-of-using-the-nativeaot-deployment) for additional details. The support in .NET 7 release is targeted towards console type applications.
+
+> [!WARNING]
+> NativeAOT deployment is an experimental feature in .NET 7.
+
+There are two ways to publish your app as NativeAOT:
+
+01. Specify the PublishAot flag directly to the dotnet publish command. See [dotnet publish](../tools/dotnet-publish.md) for details.
+
+    ```dotnetcli
+    dotnet publish -c Release -r win-x64 -p:PublishAot=true
+    ```
```

Comment:
Let's not talk about this one.

PublishAot sets PublishTrimmed to true. We decided we don't want to advertise passing it on the command line because it bypasses analyzers and other things that emulate trimmed behavior in `dotnet run`.